model-optimizer/mo/middle/passes/eliminate.py

```diff
@@ -107,7 +108,7 @@ def mark_const_producer_nodes(graph):
                 graph.node[input]['is_const_producer'] = False
                 graph.node[output]['is_const_producer'] = False
 
-        if not node.has('value') or node.value is None:
+        if not node.has('value') or node.value is None or not is_fully_defined(node.value):
```

Comment:
this is equivalent to the previous version isn't it?

Reply:
No. The node cannot be a const producer if any of the following conditions apply:
1. It does not have `value` attribute
2. The `value` is None
3. The `value` contains dynamic values. And this condition previously didn't exist.